### PR TITLE
Add null safety to example app

### PR DESCRIPTION
**Pull request by fotiDim**
What the title says 👆
Please test it before approving!

### CODE REVIEW

**Review by brim-borium — COMMENTED** [review]
The files look good but during testing I found an issue in the spotify_sdk.dart

`var capabilitiesJson= await (_channel.invokeMethod<String>(MethodNames.getCapabilities) as FutureOr<String>);`

The cast with `as FutureOr<String>` seems not to work with `_channel.invokeMethod<String>()`. This returns a `Future<String?>`

We can go two ways here, returning null or throw an exception. Not sure what's better here

```
static Future<Capabilities?> getCapabilities(
      {required String spotifyUri}) async {
    try {
      var capabilitiesJson =
          await _channel.invokeMethod<String>(MethodNames.getCapabilities);

      if (capabilitiesJson!.isNotEmpty) {
        var capabilitiesMap =
            jsonDecode(capabilitiesJson) as Map<String, dynamic>;
        return Capabilities.fromJson(capabilitiesMap);
      }

      return null;
    } on Exception catch (e) {
      _logException(MethodNames.getCapabilities, e);
      rethrow;
    }
  }

```

Same goes for all the places we use `as FutureOr<String>`

**Comment by fotiDim** [comment]
@brim-borium I hope I address what you pointed out.
It turned out 2.0 was not working for iOS. I had to fix some bugs in the null safety migration. Let's try to make a release soonish. I have one more PR coming up after this is merged.

**Comment by brim-borium** [comment]
Can this be merged then and you fix the iOS issues in a separate pr or do you want to do this here?

File: lib/models/connection_status.dart [hunk 1]
```diff
@@ -14,7 +14,7 @@ class ConnectionStatus {
   @JsonKey(name: 'connected')
   final bool connected;
   @JsonKey(name: 'message')
-  final String message;
+  final String? message;
```
Comment (fotiDim) [inline lib/models/connection_status.dart:17]: This is for iOS. On successful connection we are only sending back the `connected` bool without any message.

File: lib/models/connection_status.g.dart [hunk 2]
```diff
@@ -8,7 +8,7 @@ part of 'connection_status.dart';
 
 ConnectionStatus _$ConnectionStatusFromJson(Map<String, dynamic> json) {
   return ConnectionStatus(
-    json['message'] as String,
+    json['message'] as String?,
```
Comment (fotiDim) [inline lib/models/connection_status.g.dart:11]: I had to edit this. Not sure how it is regenerated

File: example/lib/main.dart [hunk 3]
```diff
@@ -462,7 +457,7 @@ class _HomeState extends State<Home> {
     }
   }
 
-  Future<void> setShuffle({bool shuffle}) async {
+  Future<void> setShuffle(bool shuffle) async {
```
Comment (fotiDim) [inline example/lib/main.dart:460]: This is a breaking change. I documented this in the changelog.

File: example/lib/main.dart [hunk 4]
```diff
@@ -196,89 +197,83 @@ class _HomeState extends State<Home> {
...
-                          DropdownMenuItem(
-                            value: RepeatMode.track,
-                            child: Text('track'),
-                          ),
-                          DropdownMenuItem(
-                            value: RepeatMode.context,
-                            child: Text('context'),
-                          ),
-                        ],
-                        onChanged: setRepeatMode,
```
Comment (fotiDim) [inline example/lib/main.dart:258]: How was that even working

File: example/lib/main.dart [hunk 5]
```diff
@@ -196,89 +197,83 @@ class _HomeState extends State<Home> {
...
     return StreamBuilder<PlayerState>(
       stream: SpotifySdk.subscribePlayerState(),
-      initialData: PlayerState(
-        null,
-        1,
-        1,
-        null,
-        null,
-        isPaused: false,
-      ),
```
Comment (fotiDim) [inline example/lib/main.dart:206]: I hope you don't mind removing that.

**Comment by fotiDim** [comment]
> Can this be merged then and you fix the iOS issues in a separate pr or do you want to do this here?

The iOS issues are fixed in this PR. The upcoming one is just for a new "feature".

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,6 @@
+## 2.0.1
```
Comment (brim-borium) [inline CHANGELOG.md:1]: If you add a breaking change we must raise the version to 3.0.0

File: example/lib/main.dart (hunk 5 above)
Comment (brim-borium) [reply example/lib/main.dart:206]: Fine with that :)

File: example/lib/main.dart (hunk 4 above)
Comment (brim-borium) [reply example/lib/main.dart:258]: What do you mean?

**Review by brim-borium — CHANGES_REQUESTED** [review]

File: example/lib/main.dart (hunk 4 above)
Comment (fotiDim) [reply example/lib/main.dart:258]: We were passing the function as a lambda but `setRepeatMode` requires an argument. How was it getting that? I tested it and did not work for me.

File: CHANGELOG.md (hunk 6 above)
Comment (fotiDim) [reply CHANGELOG.md:1]: It's a bummer that we spotted it now after bumping to 2.0. It is a very minor change which I don't think justifies a new major release just for that. My suggestion is that we either set it to 2.1.0 or we skip it for now and do this change later.

Comment (brim-borium) [reply CHANGELOG.md:1]: Then lets bump this to 2.1.0

File: example/lib/main.dart (hunk 4 above)
Comment (brim-borium) [reply example/lib/main.dart:258]: Actually I am not sure how that happened 🤷

**Review by brim-borium — CHANGES_REQUESTED** [review]
Lets set the version to 2.1.0 in the changelog and this should be fine

File: CHANGELOG.md (hunk 6 above)
Comment (fotiDim) [reply CHANGELOG.md]: @brim-borium done

**Review by brim-borium — APPROVED** [review]

**Pull request merged**